src/main/java/lotto/ui/output/result/GameResult.java

```diff
@@ -0,0 +1,59 @@
+package lotto.ui.output.result;
+
+import lotto.core.omr.Omr;
+import lotto.core.omr.OmrCard;
+import lotto.core.round.Rank;
+import lotto.core.round.Round;
+import lotto.ui.Program;
+import lotto.util.RateUtils;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class GameResult {
+    private final double totalPrice;
+    private final double totalWinningMoney;
+    private final Map<Rank, List<Omr>> result;
+
+    public GameResult(OmrCard omrCard, Round round) {
+        this.result = omrCard.grade(round);
+        this.totalPrice = calcTotalPrice(omrCard);
+        this.totalWinningMoney = calcTotalWinningMoney();
+    }
+
+    public double yields() {
+        return RateUtils.getRate(totalPrice, totalWinningMoney);
+    }
+
+    public int totalCount(Rank rank) {
+        if (rank.isDuplicateMatch()) {
+            return duplicateMatchCount();
+        }
+
+        return count(rank);
+    }
+
+    private int duplicateMatchCount() {
+        return Rank.duplicateMatchValues().stream()
+            .mapToInt(rank -> count(rank))
+            .sum();
+    }
+
+    private int count(Rank rank) {
+        return Optional.ofNullable(result.get(rank))
+            .orElse(Collections.EMPTY_LIST)
```

Comment:
```suggestion
            .orElse(Collections.emptyList())
```
EMPTY_LIST는 List에 제네릭 타입을 명시 하지 않아서 소나린트 경고가 뜨네요

Reply:
로타입이였네요. 감사합니다